pkg/controller/wildflyserver/wildflyserver_controller.go

```diff
@@ -116,12 +121,47 @@ func (r *ReconcileWildFlyServer) Reconcile(request reconcile.Request) (reconcile
 		return reconcile.Result{}, err
 	}
 
-	// Check if the statefulSet already exists, if not create a new one
+	var statefulSetDoesNotExists bool
 	foundStatefulSet := &appsv1.StatefulSet{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: wildflyServer.Name, Namespace: wildflyServer.Namespace}, foundStatefulSet)
 	if err != nil && errors.IsNotFound(err) {
+		statefulSetDoesNotExists = true
+	} else if err != nil {
+		reqLogger.Error(err, "Failed to get StatefulSet.")
+		return reconcile.Result{}, err
+	} else {
+		statefulSetDoesNotExists = false
+	}
+
+	// Define secret jboss cli connection for txn recovery processing
+	foundTxnRecoverySecret := &v1.Secret{}
```

Comment:
The `Secrete` is created before the `StatefulSet` creation. But all this logic comes from the way how the reconcile is expected to work. The `StatefulSet` needs the secret before it's created. If secret exists then it could be used. If it does not exist then the secret needs to be created - time for reconcile loop to be requeued. If there is existing `StatefulSet` but somebody removed the `Secret` then the new `Secret` needs to be created and the `StatefulSet` needs to be recreated as it links the non-existent `Secret` at the time.

Reply:
For code maintenance, it would  be simpler to keep the reconcile loop as linear as possible

1. check if the secret exists
   1.  if it does not exist, create it and reconcile
1. check if the StatefulSet exists
   1. if it does not exist, create it (with the info from the existing secret) and reconcile
   1. if it exists, check that its spec is consistent with the existing secret
      1. if that's not the case, delete the statefulset and reconcile
1. etc....

Reply:
Ok, I changed the code in this way.